### PR TITLE
Add functions to remove unused map in Switch

### PR DESCRIPTION
**Pull request by wenyingd**
Add functions to create Table/Group/Meter object on the Switch but not store in the db (tableDb/groupDb/meterDb). This is because the application may maintain its own cache for the Table/Group/Meter resources, and a storage in ofnet is duplicated.

Add APIs in AppInterface to control if or not to initialize the flow graph and the tlv map on the Switch.

### CODE REVIEW

File: ofctrl/ofSwitch.go [hunk 1]
```diff
@@ -108,7 +110,9 @@ func NewSwitch(stream *util.MessageStream, dpid net.HardwareAddr, app AppInterfa
 	}
 	// Prepare a context for current connection.
 	s.ctx, s.cancel = context.WithCancel(context.Background())
-	s.tlvMgr = newTLVMapMgr()
+	if app.TLVMapEnabledOnSwitch() {
```
Comment (hongliangl) [inline ofctrl/ofSwitch.go:113]: Since the return value is always true, why add an if here?

File: ofctrl/ofSwitch.go [hunk 2]
```diff
@@ -96,7 +96,9 @@ func NewSwitch(stream *util.MessageStream, dpid net.HardwareAddr, app AppInterfa
 		s.ctrlID = ctrlID
 
 		// Initialize the fgraph elements
-		s.initFgraph()
+		if app.FlowGraphEnabledOnSwitch() {
```
Comment (hongliangl) [inline ofctrl/ofSwitch.go:99]: ditto

File: ofctrl/ofctrl.go [hunk 3]
```diff
@@ -94,6 +94,10 @@ type AppInterface interface {
 
 	// Controller received a multi-part reply from the switch
 	MultipartReply(sw *OFSwitch, rep *openflow15.MultipartReply)
+
+	FlowGraphEnabledOnSwitch() bool
+
+	TLVMapEnabledOnSwitch() bool
```
Comment (hongliangl) [inline ofctrl/ofctrl.go:100]: Why adding these two methods?

File: ofctrl/ofSwitch.go (hunk 1 above)
Comment (wenyingd) [reply ofctrl/ofSwitch.go:113]: The `ofActor` in the test is an implementation of the Interface `AppInterface`. Returning true in the test code is to ensure the related code branch is able to tested. For other implementation like antrea, the return value may be false.

File: ofctrl/ofctrl.go (hunk 3 above)
Comment (wenyingd) [reply ofctrl/ofctrl.go:100]: It is to provide more options to the AppInterface to decide if the tlvMap and the flowGraph ( tableDb/groupDb/meterDb ) is used or not.

**Review by tnqn — APPROVED** [review]
LGTM

**Pull request merged**